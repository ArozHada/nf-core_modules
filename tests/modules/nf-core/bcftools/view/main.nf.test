nextflow_process {

    name "Test Process BCFTOOLS_VIEW"
    script "modules/nf-core/bcftools/view/main.nf"
    process "BCFTOOLS_VIEW"
    config "tests/modules/nf-core/bcftools/view/nextflow.config"
    tag "bcftools_view"

    test("Sarscov2 Illumina VCF") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [[ id:'out', single_end:false ], // meta map
                            file(params.test_data['sarscov2']['illumina']['test_vcf_gz'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_vcf_gz_tbi'], checkIfExists: true)
                           ]
                input[1] = []
                input[2] = []
                input[3] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("view") }
            )
        }
    }

    test("Sarscov2 Illumina VCF With Regions & Targets") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [[ id:'out', single_end:false ], // meta map
                            file(params.test_data['sarscov2']['illumina']['test_vcf_gz'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_vcf_gz_tbi'], checkIfExists: true)
                           ]
                input[1] = file(params.test_data['sarscov2']['illumina']['test3_vcf_gz'], checkIfExists: true)
                input[2] = file(params.test_data['sarscov2']['illumina']['test2_vcf_targets_tsv_gz'], checkIfExists: true)
                input[3] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("view_with_regions_targets") }
            )
        }
    }
}
