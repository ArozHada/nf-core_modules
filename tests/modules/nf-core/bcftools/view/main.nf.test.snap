{
    "view": {
        "content": [
            [
                "out.vcf.gz:md5,4f24467157f5c7a3b336481acf0c8a65",
                "versions.yml:md5,50dbfa1ac43f11ea9696bd8faa2ffde2"
            ]
        ],
        "timestamp": "2023-04-11T01:11:29+0000"
    },
    "view_with_regions_targets": {
        "content": [
            [
                "out.vcf.gz:md5,ed691eee98c5c5abdf751e3006b8e206",
                "versions.yml:md5,50dbfa1ac43f11ea9696bd8faa2ffde2"
            ]
        ],
        "timestamp": "2023-04-11T01:11:29+0000"
    }
}